File: Codigos estudiantes por lenguaje/Java/Paula Rios/ordenamientos/shellSort.java
public class Main {

    static void shellSort(int arr[]) {
        int n = arr.length;
        for (int salto = n/2; salto > 0; salto /= 2) {
            for (int i = salto; i < n; i++) {
                int temp = arr[i];
                int j;
                for (j = i; j >= salto && arr[j-salto] > temp; j-= salto) {
                    arr[j] = arr[j-salto];
                }
                arr[j] = temp;
            }
        }
    }
    
    static void print(int arr[]) {
        int n = arr.length;
        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }
    
    public static void main(String[] args) {
        int arr[] = {12, 34, 54, 2, 0, 99, -1};
        print(arr);
        shellSort(arr);
        print(arr);
        
    }

}

File: Codigos estudiantes por lenguaje/PY/Daniel Cangrejo/Ordenamientos/Shell_Sort.py
from sys import stdin, stdout
rl = stdin.readline
wr = stdout.write


def shellSort(arr=list, n=int):

    salto = n // 2
    while salto > 0:
        for i in range(salto, n):
            temp = arr[i]
            j = i
            while j >= salto and arr[j-salto] > temp:
                arr[j] = arr[j-salto]
                j -= salto
            arr[j] = temp
        salto //= 2


arr = list(map(int, rl().split()))
shellSort(arr, len(arr))
print(*arr)
